gbpservice/nfp/configurator/drivers/firewall/paloalto/paloalto_fw_driver.py

```diff
@@ -0,0 +1,875 @@
+            self._print_exception('UnexpectedError', err,
+                                  const.ADD_SECURITY_APPLICATION, resp)
+            raise Exception(err)
+    
+        msg = "POSTED application '%s' to Service VM" % application_name
+        LOG.info(msg)
+        return application_name
+
+    def create_service_element(self, service_name,
+                                     protocol='tcp',
+                                     port='80'):
+        if protocol == 'tcp':
+            string = ((const.SECURITY_SERVICE_TCP_TEMPALTE) % (service_name,
+                                                               port))
+        elif protocol == 'udp':
+            string = ((const.SECURITY_SERVICE_UDP_TEMPALTE) % (service_name,
+                                                               port))
+        return string
+
+    def create_application_element(self, application_name):
+        string = ((const.SECURITY_APPLICATION_TEMPALTE) % (application_name))
+        return string
+
+    def add_security_service(self, hostname, element):
+        xapi = self.build_xapi(hostname)
+        xpath = const.SECURITY_SERVICE_URL
+        xapi.set(xpath=xpath, element=element)
+        return self.xml_python(xapi)
+
+    def add_security_application(self, hostname, element):
+        xapi = self.build_xapi(hostname)
+        xpath = const.SECURITY_APPLICATION_URL
+        xapi.set(xpath=xpath, element=element)
+        return self.xml_python(xapi)
+
+    def delete_security_service(self, hostname, service_name):
+        xapi = self.build_xapi(hostname)
+        xpath = const.SECURITY_SERVICE_ENTRY_URL % service_name
+        xapi.delete(xpath=xpath)
+        return self.xml_python(xapi)
+
+    def delete_security_application(self, hostname, application_name):
+        xapi = self.build_xapi(hostname)
+        xpath = const.SECURITY_APPLICATION_ENTRY_URL % application_name
+        xapi.delete(xpath=xpath)
+        return self.xml_python(xapi)
+
+    def add_security_rule(self, hostname, element):
+        xapi = self.build_xapi(hostname)
+        xpath = const.SECURITY_RULES_URL
+        xapi.set(xpath=xpath, element=element)
+        return self.xml_python(xapi)
+
+    def edit_security_rule(self, hostname, rule_name, element):
+        xapi = self.build_xapi(hostname)
+        xpath = const.SECURITY_RULE_ENTRY_URL % rule_name
+        xapi.edit(xpath=xpath, element=element)
+        return self.xml_python(xapi)
+
+    def delete_security_rule(self, hostname, rule_name):
+        xapi = self.build_xapi(hostname)
+        xpath = const.SECURITY_RULE_ENTRY_URL % rule_name
+        xapi.delete(xpath=xpath)
+        return self.xml_python(xapi)
+
+    def show_security_rule(self, hostname, rule_name):
+        xapi = self.build_xapi(hostname)
+        xpath = const.SECURITY_RULE_ENTRY_URL % rule_name
+        xapi.show(xpath=xpath)
+        return self.xml_python(xapi)
+
+    def configure_healthmonitor(self, context, kwargs):
```

Comment:
Health monitor is generalized in the base driver class. Remove this and reuse from base driver class. Check vyos driver for reference.